agent/tests/test_lib_aws_sdk_php.c

```diff
@@ -6,12 +6,284 @@
+   * value. */
+  expr = nr_php_call(NULL, "one_param", first_arg);
+  tlib_pass_if_not_null("Expression should evaluate.", expr);
+  nr_php_zval_free(&expr);
+
+  /* Invalid case: no parameter.  The wrapper should see the null return value.
+   */
+  expr = nr_php_call(NULL, "no_param");
+  tlib_pass_if_not_null("Expression should evaluate.", expr);
+  nr_php_zval_free(&expr);
+
+  /*
+   *Invalid case: QueueUrl not found in the argument array.  The wrapper should
+   *see the null return value.
+   */
+  char* no_queueurl_arg
+      = "array("
+        "    0 => array("
+        "        'Nope' => 'curly_q'"
+        "    )"
+        ")";
+  array_arg = tlib_php_request_eval_expr(no_queueurl_arg);
+  expr = nr_php_call(NULL, "two_param", first_arg, array_arg);
+  tlib_pass_if_not_null("Expression should evaluate.", expr);
+  nr_php_zval_free(&expr);
+  nr_php_zval_free(&array_arg);
+
+  /*
+   *Invalid case: inner arg in the argument array is not an array.  The wrapper
+   *should see the null return value.
+   */
+  char* arg_in_array_not_array
+      = "array("
+        "    0 => '1'"
+        ")";
+  array_arg = tlib_php_request_eval_expr(arg_in_array_not_array);
+  expr = nr_php_call(NULL, "two_param", first_arg, array_arg);
+  tlib_pass_if_not_null("Expression should evaluate.", expr);
+  nr_php_zval_free(&expr);
+  nr_php_zval_free(&array_arg);
+
+  /*
+   *Invalid case: empty argument array.  The wrapper should see
+   * the null return value.
+   */
+  char* no_arg_in_array
+      = "array("
+        ")";
+  array_arg = tlib_php_request_eval_expr(no_arg_in_array);
+  expr = nr_php_call(NULL, "two_param", first_arg, array_arg);
+  tlib_pass_if_not_null("Expression should evaluate.", expr);
+  nr_php_zval_free(&expr);
+  nr_php_zval_free(&array_arg);
+
+  /*
+   *Invalid case: The argument array is not an array.  The wrapper should see
+   * the null return value.
+   */
+  char* array_arg_not_array = "1";
+  array_arg = tlib_php_request_eval_expr(array_arg_not_array);
+  expr = nr_php_call(NULL, "two_param", first_arg, array_arg);
+  tlib_pass_if_not_null("Expression should evaluate.", expr);
+  nr_php_zval_free(&expr);
+  nr_php_zval_free(&array_arg);
+
+  nr_php_zval_free(&first_arg);
+  tlib_php_request_end();
+  tlib_php_engine_destroy();
+}
+
+static inline void test_message_param_queueurl_settings_expect_val(
+    nr_segment_message_params_t message_params,
+    nr_segment_cloud_attrs_t cloud_attrs,
+    char* cloud_region,
+    char* cloud_account_id,
+    char* destination_name) {
+  tlib_pass_if_str_equal("cloud_region should match.", cloud_attrs.cloud_region,
+                         cloud_region);
+  tlib_pass_if_str_equal("cloud_account_id should match.",
+                         cloud_attrs.cloud_account_id, cloud_account_id);
+  tlib_pass_if_str_equal("destination_name should match.",
+                         message_params.destination_name, destination_name);
+
+  nr_free(cloud_attrs.cloud_region);
```

Comment:
If the parse function is changed to not strdup the values this function will need to be modified too.

Reply:
With the parse_queueurl refactor, tests need to be updated with modifiable strings and to not free the previously strduped strings.
[0b194c2](https://github.com/newrelic/newrelic-php-agent/pull/1003/commits/0b194c2382d09799451eee9274fe9ddd02531586)

Reply:
Please see: [43fa2b8](https://github.com/newrelic/newrelic-php-agent/pull/1003/commits/43fa2b888d6583004d8159325270f1de6c0e16e9)